### ISSUE TITLE
No scripts run when custom search engine is used to load Wikipedia

### ISSUE THREAD

**Issue by nicolasff** [Tampermonkey/tampermonkey#1650]
Google Chrome has a feature called "site search" where users can define a custom keyword to construct a URL containing a search term, in effect directly accessing the search results page of a website. When this feature is used to load a Wikipedia page, no Tampermonkey scripts will run on that page even once it's fully loaded. The extension menu shows "No script is running".

## Expected Behavior

Scripts should run as if the full URL had been typed in manually.

## Actual Behavior

No scripts run.

## Specifications

- Google Chrome: 107.0.5304.107
- TM: 4.18.0
- OS: macOS 12.6 (21G115)

## Script

If needed, save the following script that runs on all Wikipedia pages and logs a single line to the console. It should run on all Wikipedia pages, and if it's not listed on a fully loaded page then you've reproduced the issue.

```js
// ==UserScript==
// @name         Wikipedia Hello World
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  Prints a single line in the console
// @author       https://github.com/nicolasff
// @match        https://*.wikipedia.org/*
// @icon         https://en.wikipedia.org/favicon.ico
// @grant        none
// ==/UserScript==

(function() {
    console.log('Hello, Wikipedia!');
})();
```

## Steps to reproduce

Add a custom search engine to search Wikipedia, let's call it `wiki`:
1. Open the Google Chrome settings
2. Click on "Search engine" on the left
3. Click on "Manage search engines and site search" in the middle
4. Scroll down if needed, to the "Site search" category and click the blue "Add" button on the right
5. Add the Wikipedia search engine:
    1. In "Search Engine", enter `Wikipedia`
    2. In "Shortcut", enter `wiki`
    3. In "URL with %s in place of query", enter `https://en.wikipedia.org/w/index.php?search=%s`
    4. Click the "Add" button to save
6. Close the settings tab

Try a custom Wikipedia search:
1. Pin Tampermonkey to keep its icon visible
2. Open a new tab
3. In the URL bar, type `wiki` then add a space
4. The input field should now show "Search Wikipedia" left of your cursor
6. Type `Earth` and press Enter
7. The search results page will load and will redirect you to the Wikipedia article for Earth at `https://en.wikipedia.org/wiki/Earth`

The article page fully loads, but **no scripts are run**. The red badge with the number of scripts run is missing.

Interestingly with Wikipedia when a page is first loaded this way, even after clicking on an internal link to load a different article the scripts will still not start running. If you're on the article for "Earth" from the reproduction steps and click on "Sun", that article will load but the script count badge will still not appear.

Force-reloading the page will cause the scripts to start.

This seems to have started happening somewhat recently, in the past couple of months maybe. I've been using both Tampermonkey and a custom search engine for Wikipedia for years and the two used to work just fine together.

**Comment by derjanb** [comment]
Hmm, this is working fine here. Tested with:

* Ubuntu with Version 108.0.5359.19 dev and Version 107.0.5304.68 (Official Build) beta
* Window with Version 106.0.5249.103 and Version 107.0.5304.107

Can you please try with a new temporary profile?

**Comment by nicolasff** [comment]
After several months hitting this issue relatively often, I finally managed to reproduce it in a tab with a console open.

I noticed the "Uncaught pagejs missing" error that was logged, with a link to [question 208 of the FAQ](https://www.tampermonkey.net/faq.php#Q208), which mentioned this is due to a Chromium bug.

I went through the resolution steps (disable/restart/enable/restart), and this seems to have resolved the issue.
